druid/src/core.rs

```diff
@@ -377,7 +380,18 @@ impl<T: Data, W: Widget<T>> WidgetPod<T, W> {
             is_root: false,
             focus_widget: ctx.focus_widget,
         };
-        let rect = child_ctx.base_state.layout_rect;
+
+        let rect = match child_ctx.base_state.layout_rect {
+            Some(rect) => rect,
+            None => {
+                log::warn!(
+                    "WidgetPod received an event without having been laid out. \
+                This likely indicates a missed call to set_layout_rect."
+                );
+                Rect::ZERO
+            }
+        };
```

Comment:
How about
```suggestion
        if child_ctx.base_state.layout_rect.is_none() {
            log::warn!(
                "WidgetPod received an event without having been laid out. \
                This likely indicates a missed call to set_layout_rect."
            );
        }
        let rect = child_ctx.base_state.layout_rect.unwrap_or_default();
```

Reply:
I don't think I have clearly prefer your version, but since you did all that typing I'm happy to take it. ;)